### ISSUE TITLE
[Zoe] OfferRules.offer is a confusing name

### ISSUE THREAD

**Issue by katelynsills** [Agoric/agoric-sdk#683]
@Chris-Hibbert [said](https://github.com/Agoric/agoric-sdk/commit/3934c71ee4c1545db8e6ea12e33dc6c08ea5bdc6#r37813792):

> There's a potential confusion between the offer's rules and the offer clause in the rules. ...we may want to consider renaming one of them.

To explain further, we call the entire structure "offerRules" and offerRules has a key "offer" that is what the person is offering. 

Update (3/18): 

In this morning's meeting, we decided to retain the name "offer" for the combination of the offerRules, payments, and invite as arguments to `zoe.redeem`. 

We are changing `offerRules` to `proposal`.
We are changing `offerRules.offer` to `give`
`want` and `exit` in the `offerRules` (now `proposal`) remain the same. 

In summary: A user makes an `offer` with Zoe. Their offer has a `proposal` with the following structure:
```js
{
    give: { },
    want: { },
    exit: { },
}
``` 

This Issue should be closed in #685

**Comment by Chris-Hibbert** [comment]
To be even more explicit, offerRules (currently) have two keys: 'offer' and 'want'. We need a pair of terms, plus a term for the overall structure.

**Comment by katelynsills** [comment]
Closed by #685
